src/js/actions/tool/type.js

```diff
@@ -0,0 +1,263 @@
+                            layers = Immutable.List.of(layer);
+
+                        this.flux.actions.type.updatePropertiesThrottled(document, layers, properties);
+                    });
+            }
+        }.bind(this);
+        descriptor.addListener("createTextLayer", _layerCreatedHandler);
+
+        _layerDeletedHandler = function (event) {
+            _layerDeleted = event.layerID;
+            this.flux.actions.typetool.handleDeletedLayer(event, _layersReplaced);
+        }.bind(this);
+        descriptor.addListener("deleteTextLayer", _layerDeletedHandler);
+
+        _toolModalStateChangedHandler = function (event) {
+            if (event.kind._value === "tool" && event.tool.ID === "txBx" &&
+                event.state._value === "exit" && event.reason._value === "cancel") {
+                // If there was a deleteTextLayer event, we've already updated the model.
+                if (_layerDeleted) {
+                    _layerDeleted = null;
+                    return;
+                }
+
+                this.flux.actions.typetool.handleTypeModalStateCanceled();
+            }
+        }.bind(this);
+        descriptor.addListener("toolModalStateChanged", _toolModalStateChangedHandler);
+
+        return UI.setSuppressTargetPaths(true)
+            .bind(this)
+            .then(function () {
+                if (firstLaunch) {
+                    firstLaunch = false;
+
+                    return descriptor.playObject(toolLib.resetTypeTool("left", "Myriad Pro", 16, [0, 0, 0]));
+                }
+            });
+    };
+    select.reads = [];
+    select.writes = [locks.PS_APP, locks.PS_TOOL];
+    select.transfers = [];
+    select.modal = true;
+
+    /**
+     * Remove modal type event handlers.
+     *
+     * @return {Promise}
+     */
+    var deselect = function () {
+        descriptor.removeListener("move", _moveHandler);
+        descriptor.removeListener("updateTextProperties", _typeChangedHandler);
+        descriptor.removeListener("createTextLayer", _layerCreatedHandler);
+        descriptor.removeListener("deleteTextLayer", _layerDeletedHandler);
+        descriptor.removeListener("toolModalStateChanged", _toolModalStateChangedHandler);
+        
+        _moveHandler = null;
+        _typeChangedHandler = null;
+        _layerCreatedHandler = null;
+        _layerDeletedHandler = null;
+        _toolModalStateChangedHandler = null;
+
+        return UI.setSuppressTargetPaths(false);
+    };
+    deselect.reads = [];
+    deselect.writes = [locks.PS_APP];
+    deselect.transfers = [];
+    deselect.modal = true;
+
+    /**
+     * Handles updateTextProperties events, which are emitted during the type modal
+     * tool state to indicate the text properties of the current text selection or
+     * cursor position.
+     *
+     * @param {object} event
+     * @return {Promise}
+     */
+    var updateTextPropertiesHandler = function (event) {
```

Comment:
This was a helper function used by the type tool and the select type subtool, but now it's an action.